src/main/java/com/prajina/academy/model/Course.java

```diff
@@ -0,0 +1,100 @@
+package com.prajina.academy.model;
+
+import java.util.Objects;
+
+/**
+ * This class represent the courses
+ * 
+ * @author Prajina
+ *
+ */
+/**
+ * @author prajina.purayil
+ *
+ */
+public class Course {
+	// TODO move common fields to base class
+	private long id;
```

Comment:
you should really use the Long (wrapper class), because you want to have a null, for use in empty forms